articles/ToDoList-with-Flask-Firebase/index.md

```diff
@@ -0,0 +1,241 @@
+
+### Prerequisites
+
+- Basic knowledge of Python and HTML.
+
+- [Python ](https://www.python.org/downloads/) installed in the system.
+
+- Flask installed in the system (You can refer this [article](https://www.section.io/engineering-education/python-frameworks/) for a gentle introduction and installation instructions)
+
+### Step 1: Basic HTML template
+
+1. Create a folder called ‘templates’ within the project folder. We store our HTML files in this folder. 
+
+**Note**: You cannot give this folder a different name because of the conventions specified by Flask. If it is not named as ‘templates’, then the flask server will not detect the HTML files.
+
+2. Now open an HTML file named ‘index.html’ in a text editor of your choice. Since the objective of this tutorial is to learn to use Flask and Firebase Database, let us build a very simple UI. Add the HTML code to create a page that looks like this - 
+
+![img](./webpage1.png)
+
+```html
+<!DOCTYPE html>
+<html lang="en" dir="ltr">
+  <head>
+    <meta charset="utf-8">
+    <title>To-Do List</title>
+    <h1>To-Do List</h1>
+  </head>
+  <body>
+    <div>
+      <form id="addTask" action="update_task" method="post">
+        <fieldset>
+          <h3>Add Task</h3>
+          <input placeholder="Paste Here" name='Task' type="text"  required >
+          </input>
+          <button class="green" name="submit" type="submit" id="addTask" style="float: right;">Add</button>
+        </fieldset>
+      </form>
+    </div>
+    <br><br>
+<!-- Task List -->
+    <div style="background-color: lightcyan">
+        <fieldset>
+          <form id="deleteTask" action="{{ url_for('delete_task') }}" method="post">
+          <button  name="delete" type="submit"  style="float: right;">Delete</button>
+          </form>
+          <form id="updateTask" action="{{ url_for('update_task') }}" method="post">
+          <input type="text" name="task" value=’Study Chapter 1’>
+          <br>
+          <button  name="update" type="submit"  style="float: left;">Update</button>
+          </form>
+        </fieldset>
+    </div>
+<!-- Task List -->
+  </body>
+</html>
+```
+
+### Step 2: Set up Firebase Realtime Database
+
+1. We will be using the Python firebase package - Pyrebase, so install this into your python environment.
+
+   `pip install pyrebase`
+
+2. Log in to https://console.firebase.google.com/.
+
+3.  Click on Add project 
+
+4. Name your project
+
+   ![img](./fb1.png)
+
+5. We won’t be using Google Analytics for this project, so you can disable it.
+
+6. Finally, click on Create Project. 
+
+7. Go to Realtime Database and click on Create Database. Enable test mode, which allows anyone who has the tokens of this project to read and write to the database. 
+
+   Ensure that your rules look like this, for now. You can always modify it later based on your requirements.
```

Comment:
Consider changing
"You can always modify it later..." to "You can always modify them later..."
Because you are referring to the rules if I'm not mistaken.